include/lug/Window/Window.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <memory>
+#include <queue>
+#include <lug/Window/Export.hpp>
+
+namespace lug {
+namespace Window {
+namespace priv {
+    class WindowImpl;
+} // namespace priv
+
+enum class LUG_WINDOW_API EventType
+{
+    CLOSE
+};
+
+struct LUG_WINDOW_API Event
+{
+    lug::Window::EventType type;
+};
+
+enum class LUG_WINDOW_API Style
```

Comment:
You need to define bitwise operations, for example this code won't work.
`Style style = Style::Titlebar | Style::Resize;`